java/src/main/java/com/adhoc/flight/utils/QueryUtils.java

```diff
@@ -0,0 +1,241 @@
+package com.adhoc.flight.utils;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import org.apache.arrow.util.Preconditions;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.ipc.ArrowStreamWriter;
+
+public final class QueryUtils {
+
+    private QueryUtils() {
+        // Prevent instantiation.
+    }
+
+    public static final class Writer {
```

Comment:
Why make this an inner class when it just has static methods?

Reply:
I agree with you, I over-engineered some aspects of this code. Hopefully fixed now.